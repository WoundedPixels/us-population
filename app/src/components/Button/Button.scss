@import '../../styles/colors.scss';

.Button {
  padding: 5px 10px;
  color: $lightGray;
}
