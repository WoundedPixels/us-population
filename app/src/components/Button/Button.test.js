import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

const fpp = '';

it('no callback matches snapshot', () => {
  const tree = renderer.create(<Button>Hi</Button>).toJSON();
  expect(tree).toMatchSnapshot();

  const someOtherJSON = { flavor: 'fudge' };
  expect(someOtherJSON).toMatchSnapshot();
});

it('with callback matches snapshot', () => {
  const cb = () => {};
  const tree = renderer.create(<Button onClick={cb}>Hi</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('arbitrary json matches snapshot', () => {
  const someOtherJSON = { flavor: 'fudge' };
  expect(someOtherJSON).toMatchSnapshot();
});
