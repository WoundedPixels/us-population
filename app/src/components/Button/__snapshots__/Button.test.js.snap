// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arbitrary json matches snapshot 1`] = `
Object {
  "flavor": "fudge",
}
`;

exports[`no callback matches snapshot 1`] = `
<button
  className="Button"
  onClick={undefined}
>
  Hi
</button>
`;

exports[`no callback matches snapshot 2`] = `
Object {
  "flavor": "fudge",
}
`;

exports[`with callback matches snapshot 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  Hi
</button>
`;
